src/toil/batchSystems/kubernetes.py

```diff
@@ -441,13 +445,16 @@ def issueBatchJob(self, jobNode):
             pod_spec = kubernetes.client.V1PodSpec(containers=[container],
                                                    volumes=volumes,
                                                    restart_policy="Never")
+            # Make metadata to labl the job/pod with info.
+            metadata = kubernetes.client.V1ObjectMeta(name=jobName
+                                                    ,labels={"toil_run": self.runID})
+            
             # Wrap the spec in a template
-            template = kubernetes.client.V1PodTemplateSpec(spec=pod_spec)
+            template = kubernetes.client.V1PodTemplateSpec(spec=pod_spec, metadata=metadata)
+            
             # Make another spec for the job, asking to run the template with no backoff
             job_spec = kubernetes.client.V1JobSpec(template=template, backoff_limit=0)
-            # Make metadata to tag the job with info.
-            # We use generate_name to ensure a unique name
-            metadata = kubernetes.client.V1ObjectMeta(name=jobName)
+            
             # And make the actual job
             job = kubernetes.client.V1Job(spec=job_spec,
                                           metadata=metadata,
```

Comment:
It seems a bit weird to reuse the same V1ObjectMeta instance for the job and the pod, but it seems to be working.

Reply:
Should we use different metadata and labels for job and pods? I don't think it is necessary to since it is serving one purpose which is for easy aggregation. However I do see the need if we want to log out details for each jobs and pods.

Reply:
I think it's fine unless we catch it causing a bug.